installer/install.sh

```diff
@@ -0,0 +1,246 @@
+    fi
+
+    NGINX_LISTEN_DIRECTIVE_IPV4="$(get_nginx_sed_directive 4)"
+    NGINX_LISTEN_DIRECTIVE_IPV6="$(get_nginx_sed_directive 6)"
+}
+
+prepare_installation_location() {
+    echo "Now installing LuckPermsWeb..."
+    echo
+
+    if [ ! -d "$BASE_DIR" ]; then
+        sudo mkdir "$BASE_DIR"
+        sudo chown "$USER:$GROUP" "$BASE_DIR"
+    fi
+
+    cd "$BASE_DIR"
+}
+
+install_bytebin() {
+    echo "Installing bytebin..."
+
+    mkdir -p bytebin
+    pushd bytebin > /dev/null
+
+    # Find free port (stop the service if running)
+    sudo systemctl stop bytebin.service 2> /dev/null
+    BYTEBIN_PORT="$(find_free_port "$BYTEBIN_IP" "$BYTEBIN_PORT")"
+
+    # Download and Copy the Files
+    wget -q --show-progress --progress=dot:mega https://ci.lucko.me/job/bytebin/lastSuccessfulBuild/artifact/target/bytebin.jar
+    echo
+    jq \
+        --arg ip "$BYTEBIN_IP" \
+        --argjson port "$BYTEBIN_PORT" \
+        '.host = $ip | .port = $port' \
+        "$INSTALLER_DIR/files/bytebin/config.json" > config.json
+    sed \
+        -e "s@<PATH>@$BASE_DIR/bytebin@g" \
+        -e "s/<USER>/$USER/g" \
+        -e "s/<GROUP>/$GROUP/g" \
+        "$INSTALLER_DIR/files/bytebin/bytebin.service" | sudo dd of=/etc/systemd/system/bytebin.service 2> /dev/null
+
+    # Enable and (Re)Start the Service
+    sudo systemctl daemon-reload
+    sudo systemctl enable --now bytebin.service
+
+    popd > /dev/null
+
+    echo
+}
+
+install_webfiles() {
+    echo "Installing web files..."
+    echo
+
+    pushd "$REPO_DIR" > /dev/null
+
+    # Configure web application
+    jq --arg url "$BYTEBIN_URL" '.bytebin_url = $url' config.json > config.json.tmp
+    mv -f config.json.tmp config.json
+
+    # Render webfiles
+    npm install
+    npm run build
+
+    popd > /dev/null
+
+    rm -rf webfiles
+    mv "$REPO_DIR/dist/" webfiles
+}
+
+generate_https_cert() {
+    # We don't need to generate the HTTPS certificate if it already exists
+    [ -f "$HTTPS_CERT_PATH" ] && [ -f "$HTTPS_KEY_PATH" ] && return 0
+
+    echo "Generating HTTPS certificate"
+    echo
+
+    local nginx_config_file="/etc/nginx/sites-enabled/certbot_helper_$RANDOM.conf"
+    local certdir="$BASE_DIR/webfiles"
+
+    # Configure nginx for the cerbot
+    create_nginx_file \
+        "$INSTALLER_DIR/files/nginx/luckpermsweb_header_http.conf" \
+        "$INSTALLER_DIR/files/nginx/luckpermsweb_footer_certbot.conf" | sudo dd of="$nginx_config_file" 2> /dev/null
+
+    ## Reload nginx
+    sudo nginx -t && sudo nginx -s reload
+
+    # Get certificate
+    sudo certbot certonly --webroot --rsa-key-size 4096 -w "$certdir" -d "$EXTERNAL_ADDRESS"
+
+    sudo rm -rf "$nginx_config_file" "$certdir/.well-known"
+}
+
+configure_nginx() {
+    echo "Setting up nginx..."
+    echo
+
+    pushd /etc/nginx > /dev/null
+
+    # Create config file
+    local nginx_config_file="sites-available/luckpermsweb_$EXTERNAL_ADDRESS.conf"
```

Comment:
On centos this is `/etc/nginx/conf.d/luckpermsweb_whatever.conf`

Reply:
As mentioned in the Readme it’s made for Debian based systems.  
Detecting the OS was a bit overboard I thought.  
Though that can be added later.